recipes/NNPOps/build.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+rm -rf build || true
+
+# function for facilitate version comparison; cf. https://stackoverflow.com/a/37939589
+function version2int { echo "$@" | awk -F. '{ printf("%d%02d\n", $1, $2); }'; }
+
+# adapted from https://github.com/conda-forge/faiss-split-feedstock/blob/master/recipe/build-lib.sh
+declare -a CUDA_CONFIG_ARGS
+
+# the following are all the x86-relevant gpu arches; for building aarch64-packages, add: 53, 62, 72
+ARCHES=(52 60 61 70)
+# cuda 11.0 deprecates arches 35, 50
+DEPRECATED_IN_11=(35 50)
+if [ $(version2int $cuda_compiler_version) -ge $(version2int "11.1") ]; then
+    # Ampere support for GeForce 30 (sm_86) needs cuda >= 11.1
+    LATEST_ARCH=86
+    # ARCHES does not contain LATEST_ARCH; see usage below
+    ARCHES=( "${ARCHES[@]}" 75 80 )
+elif [ $(version2int $cuda_compiler_version) -ge $(version2int "11.0") ]; then
+    # Ampere support for A100 (sm_80) needs cuda >= 11.0
+    LATEST_ARCH=80
+    ARCHES=( "${ARCHES[@]}" 75 )
+elif [ $(version2int $cuda_compiler_version) -ge $(version2int "10.0") ]; then
+    # Turing support (sm_75) needs cuda >= 10.0
+    LATEST_ARCH=75
+    ARCHES=( "${DEPRECATED_IN_11[@]}" "${ARCHES[@]}" )
+fi
+for arch in "${ARCHES[@]}"; do
+    CMAKE_CUDA_ARCHS="${CMAKE_CUDA_ARCHS+${CMAKE_CUDA_ARCHS};}${arch}-real"
+done
+# for -real vs. -virtual, see cmake.org/cmake/help/latest/prop_tgt/CUDA_ARCHITECTURES.html
+# this is to support PTX JIT compilation; see first link above or cf.
+# devblogs.nvidia.com/cuda-pro-tip-understand-fat-binaries-jit-caching
+CMAKE_CUDA_ARCHS="${CMAKE_CUDA_ARCHS+${CMAKE_CUDA_ARCHS}}"
```

Comment:
Don't you want to add `${LATEST_ARCH}-real;${LATEST_ARCH}-virtual;` here?

Reply:
That should be taken care of in line 34 on that file

Reply:
I don't understand. Can you explain more?

Reply:
Sorry line 32 is where those variables are set.

Reply:
Line 32 doesn't have the latest arch in it nor the virtual arch.

Reply:
```
2022-01-24T16:56:16.6221945Z + CMAKE_CUDA_ARCHS='52-real;60-real;61-real;70-real;75-real;80-real'
2022-01-24T16:56:16.6225902Z + CUDA_CONFIG_ARGS+=(-DCMAKE_CUDA_ARCHITECTURES="${CMAKE_CUDA_ARCHS}")
```
It looks like it correctly builds the list of arch, according to 
https://cmake.org/cmake/help/latest/prop_tgt/CUDA_ARCHITECTURES.html

>An architecture can be suffixed by either -real or -virtual to specify the kind of architecture to generate code for. If no suffix is given then code is generated for both real and virtual architectures.

So I will drop the `-real` and that should build the arch list

Reply:
Still for 11.1 you have,

`52;60;61;70;75;80`

instead of

`52;60;61;70;75;80;86`.

Also, what you have right now is more expensive in terms of compilation time and binary size.
i.e. it's equivalent to
'52-real;60-real;61-real;70-real;75-real;80-real;52-virtual;60-virtual;61-virtual;70-virtual;75-virtual;80-virtual;'

But all you need is,

'52-real;60-real;61-real;70-real;75-real;80-real;86-real;86-virtual'

Reply:
```suggestion
CMAKE_CUDA_ARCHS="${CMAKE_CUDA_ARCHS};${LATEST_ARCH}-real;${LATEST_ARCH}-virtual"
```

Reply:
duh, thanks! don't know what I was thinking :shrug: